lib/shopify_cli/theme/dev_server/app_extension_local_assets.rb

```diff
@@ -6,24 +6,26 @@ module ShopifyCLI
   module Theme
     module DevServer
       class AppExtensionLocalAssets < LocalAssetsBase
-        TAE_ASSET_REGEX = %r{(http:|https:)?//cdn\.shopify\.com/extensions/.+?/(assets/.+?\.(?:css|js))}
+        URL_REGEX = %r{(http:|https:)?//cdn\.shopify\.com/extensions/.+?/(assets/.+?\.(?:css|js))}
 
         def initialize(ctx, app, extension:)
           super(ctx, app, target: extension)
+          @base_regex = /<(script|link).*?data-app-id=\"#{extension.id}\".*?>/
         end
 
         private
 
         def replace_asset_urls(body)
-          replaced_body = body.join.gsub(TAE_ASSET_REGEX) do |match|
-            path = Regexp.last_match[2]
-            if @target.static_asset_paths.include?(path)
-              "/#{path}"
-            else
-              match
+          replaced_body = body.join.gsub(@base_regex) do |asset_tag|
```

Comment:
I believe we could keep a single `.gsub` call here with the regulat expression:

<img width="1272" alt="image" src="https://user-images.githubusercontent.com/1079279/182315580-9b2d62aa-69da-4b2f-848b-b545a11a8aa5.png">